pkg/asset/machines/gcp/gcpmachines.go

```diff
@@ -0,0 +1,112 @@
+// Package gcp generates Machine objects for gcp.
+package gcp
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types"
+	gcptypes "github.com/openshift/installer/pkg/types/gcp"
+)
+
+// GenerateMachines returns manifests and runtime objects to provision nodes using CAPI.
+func GenerateMachines(clusterID string, region string, subnet string, pool *types.MachinePool, role string, projectID string, userTags map[string]string, userLabels map[string]string) ([]*asset.RuntimeFile, error) {
+	if poolPlatform := pool.Platform.Name(); poolPlatform != gcptypes.Name {
+		return nil, fmt.Errorf("non-GCP machine-pool: %q", poolPlatform)
+	}
+	mpool := pool.Platform.GCP
+
+	total := int64(1)
+	if pool.Replicas != nil {
+		total = *pool.Replicas
+	}
+
+	var osImage string
+	if mpool.OSImage != nil {
+		osImage = mpool.OSImage.Name
+	}
+
+	// Set SecureBoot
+	shieldedInstanceConfig := capg.GCPShieldedInstanceConfig{}
+	if mpool.SecureBoot == string(machineapi.SecureBootPolicyEnabled) {
+		shieldedInstanceConfig.SecureBoot = capg.SecureBootPolicyEnabled
+	}
+
+	// Set ServiceAccount
+	instanceServiceAccount := fmt.Sprintf("%s-%s@%s.iam.gserviceaccount.com", clusterID, role[0:1], projectID)
+	// TODO - create a different service account for an xpn installation similar to current implementation ?
+
+	objectLabels := make(map[string]string, 1)
```

Comment:
unsolicited suggestion/bikeshedding: I saw `objectLabels := map[string]string{}` while reviewing and it has now become my favorite way to initialize a map if I don't care about the size.

Reply:
Good idea